src/controller/python/chip/yaml/parser.py

```diff
@@ -29,11 +29,170 @@
 from chip.yaml.errors import ParsingError, UnexpectedParsingError
 from .data_model_lookup import *
 import chip.yaml.format_converter as Converter
+from .response_storage import ResponseStorage
 
 _SUCCESS_STATUS_CODE = "SUCCESS"
+_NODE_ID_DEFAULT = 0x12345
+_ENDPOINT_DETAULT = ''  # TODO why is this an empty string
+_CLUSTER_DEFAULT = ''
+_TIMEOUT_DEFAULT = 90
 logger = logging.getLogger('YamlParser')
 
 
+@dataclass
+class _ReferenceObjects:
+    # Data model lookup to get python attribute, cluster, command object.
+    data_model_lookup: DataModelLookup = None
+    # Where various test action response are stored and loaded from.
+    response_storage: ResponseStorage = None
+    # Top level configuration values for a yaml test.
+    config_values: dict = None
+
+
+class _ConstraintValue:
+    def __init__(self, value, field_type, reference_objects: _ReferenceObjects):
+        self._response_storage = reference_objects.response_storage
+        self._load_expected_response_key = None
+        self._value = None
+
+        if value is None:
+            # Default values set above is all we need here.
+            return
+
+        if isinstance(value, str) and self._response_storage.is_key_saved(value):
+            self._load_expected_response_key = value
+        else:
+            self._value = Converter.convert_yaml_type(
+                value, field_type, reference_objects.config_values)
+
+    def get_value(self):
```

Comment:
Please describe that this gets the current value of the constraint, accounting for runtime saving of values from the DUT.